Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -3477,6 +3477,30 @@ bool GraphicsLayerCA::appendToUncommittedAnimations(const KeyframeValueList& val
     return true;
 }
 
+static const TransformOperations& transformationAnimationValueAt(const KeyframeValueList& valueList, unsigned i)
+{
+    return static_cast<const TransformAnimationValue&>(valueList.at(i)).value();
+}
+
+static bool hasBigRotationAngle(const KeyframeValueList& valueList, const SharedPrimitivesPrefix& prefix)
+{
+    const auto& primitives = prefix.primitives();
+    for (unsigned animationIndex = 0; animationIndex < primitives.size(); ++animationIndex) {
+        TransformOperation::OperationType type = primitives[animationIndex];
+        if (type != TransformOperation::ROTATE && type != TransformOperation::ROTATE_3D)
+            continue;
+        for (size_t i = 1; i < valueList.size(); ++i) {
+            auto prevOperation = static_cast<const RotateTransformOperation*>(transformationAnimationValueAt(valueList, i - 1).at(animationIndex));
```

Comment:
This should use downcast<> (as should the next line).

It's hard to grok the relationship between valueList and prefix, and I can't tell how you know that the previous operation is a rotate operation.

Reply:
I've switched this to use `downcast<>` and also have added a comment giving a bit more background about why the prefix used here.